test/framework/run.py

```diff
@@ -193,6 +194,24 @@ def test_run_shell_cmd_basic(self):
             self.assertTrue(isinstance(res.output, str))
             self.assertTrue(res.work_dir and isinstance(res.work_dir, str))
 
+    def test_fileprefix_from_cmd(self):
+        """test simplifications from fileprefix_from_cmd."""
+        cmds = {
+            'abd123': 'abd123',
+            'ab"a': 'aba',
+            'a{:$:S@"a': 'aSa',
```

Comment:
Let's add some more test cases here, like:

* `git clone` should map to `gitclone` (or maybe we want `git-clone`?)
* `cmd-with-dash` should map to `cmd-with-dash` (requires changes in `fileprefix_from_cmd`)
* `cmd_with_underscore` should map to `cmd_with_underscore` (requires changes in `fileprefix_from_cmd`)

Reply:
If the command is `git clone` then only the `git` part is passed into `fileprefix_from_cmd` and I object to adding extra code to this that increases the complexity.

Reply:
cb84bd2 (with the typo fixed in a309aae)